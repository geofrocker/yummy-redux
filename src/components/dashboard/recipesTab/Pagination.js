import React from 'react';

const Pagination = (props) => {
  const { nextPage, previousPage } = props;
  props = props.recipeData;
  return (
    <div className="col-xs-11 col-sm-4 pull-right" id="pagination">
      <ul className="pagination">
        <li className={props.disablePrevious}>
          <a className="page-link" onClick={() => previousPage(props)}>Previous</a>
        </li>
        <li className={props.disableNext}>
          <a className="page-link" onClick={() => nextPage(props)}>Next</a>
          <a className="page-link">Showing {props.page} of {props.total_pages}</a>
        </li>

      </ul>
    </div>
  );
};
export default Pagination;
