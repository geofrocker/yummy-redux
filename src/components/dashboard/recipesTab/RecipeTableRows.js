import React from 'react';
import { Thumbnail } from 'react-bootstrap';

function truncate(str, no_words) {
  return `${str.split(' ').splice(0, no_words).join(' ')}...`;
}

const RecipeTableRows = props => (
  <div className="col-md-4">
    <Thumbnail src="https://mperumpy.sirv.com/Images/recipes3.jpeg" alt="242x200">
      <h3>{props.title}</h3>
      <p>{truncate(props.ingredients, 3)}</p>
      <p>
        <a href={`/recipe/${props.recipe_id}`} className="btn btn-default"><span className="fa fa-eye"> View</span></a>&nbsp;
        <button onClick={() => props.handleEditData(props.recipe_id)} className="btn btn-primary"><span className="fa fa-edit"> Edit</span></button>&nbsp;
        <button onClick={() => props.deleteHandler(props.recipe_id)} className="btn btn-danger"><span className="fa fa-trash-alt"> Delete</span></button>
      </p>
    </Thumbnail>
  </div>
);
export default RecipeTableRows;
