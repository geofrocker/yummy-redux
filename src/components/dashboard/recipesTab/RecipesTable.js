import React, {Component} from 'react'
import PropTypes from 'prop-types';
import {connect} from 'react-redux'
import {loadRecipes, loadRecipe, loadCategories, updateRecipe, deleteRecipe, addRecipe} from '../../../actions/recipesActions'
import url from '../../../config'
import RecipeTableRows from './RecipeTableRows'
import Pagination from './Pagination'
import RecipeModal from './RecipeModal'
import toastr from 'toastr';
import { Redirect } from 'react-router-dom';

export class RecipesTable extends Component{
  constructor(props){
    super(props)
    this.state = {showMessage:false, q:'', page:1, url: url+'myrecipes?page=1',show: false,selected:'',modalTitle:'Add Recipe',recipeId:'',
    recipeData:{title:'',category:'',ingredients:'',steps:'',status:'',category_rel:''},redirect: false
    };
  }
  componentDidMount(){
    this.props.loadRecipes(this.state.url)
    .catch(xhr =>{
      xhr.response.status ===500 || !xhr.response?this.setState({ redirect: true }): this.setState({ redirect: false })
    })
    this.props.loadCategories(`${url}category?page=1`)
  }
  
  addRecipe = (e) =>{
    e.preventDefault()
    this.props.addRecipe(this.state.recipeData)
    .then(() =>{
      toastr.success('Recipe Created Successfully')
      this.setState({recipeData:{title:'',category:'',ingredients:'',steps:'',status:''}})
      this.props.loadRecipes(`${url}myrecipes?page=1`)
      this.handleClose()
    }).catch(xhr =>{
      toastr.error(this.props.message)
    })
  }
  handleRecipeUpdate = (e) => {
    e.preventDefault()
    this.props.updateRecipe(this.state.recipeId,this.state.recipeData)
    .then(() =>{
      toastr.success('Recipe Updated Successfully')
      this.setState({recipeData:{title:'',category:'',ingredients:'',steps:'',status:''}})
      this.props.loadRecipes(`${url}myrecipes?page=${this.props.page}`)
      this.handleClose()
    }).catch(xhr =>{
      toastr.error(this.props.message)
    })
  }

  deleteHandler = (id,e) =>{
    this.props.deleteRecipe(id)
    .then(() =>{
      toastr.success('Recipe Deleted Successfully')
      this.props.loadRecipes(`${url}myrecipes?page=${this.props.page}`)
    }).catch(() =>{
      toastr.error(this.props.message)
    })
  }

  handeChange = (e) => {
    const field = e.target.id
    let recipeData = this.state.recipeData
    recipeData[field] = e.target.value
    this.setState({ recipeData })
  }

  handleClose = () =>{
    this.setState({ show: false });
  }
  handleEditData = (id) =>{
    this.props.loadRecipe(id).then(() =>{
      this.setState({
        recipeData:{
          title:this.props.recipe.title,
          category:this.props.recipe.category,
          ingredients:this.props.recipe.ingredients,
          steps:this.props.recipe.steps,
          status:this.props.recipe.status,
          category_rel:this.props.recipe.category_rel
        },
        modalTitle:'Edit Recipe',
        recipeId:id
      })
    })
    this.setState({ show: true });
  }

  handleShow = () =>{
    this.setState({show:true, recipeData:{title:'',category:'',ingredients:'',steps:'',status:''},selected:'selected'});
  }

  //Handle the link to the next page from pagination
  nextPage =(props) =>{
      if (props.has_next){
          props.loadRecipes(props.next_page)
          .catch(xhr=>{
            this.setState({ redirect: true });
          })
      }
  }

  // Handle the link to the previous page from pagination
  previousPage =(props) =>{
      if(props.has_prev){
          props.loadRecipes(props.previous_page)
          .catch(xhr=>{
            this.setState({ redirect: true });
          })
      }
  };
  // Handle the search functionality
  handleRecipeSearch = (event) => {
    event.preventDefault();
    this.setState({showMessage:false,q: event.target.value,page:1});
    if(this.state.q){
        let localurl = url +'myrecipes?q='+this.state.q+'&page='+this.state.page
        this.props.loadRecipes(localurl)
    }else{
        let localurl = url +'myrecipes?page='+this.state.page
        this.props.loadRecipes(localurl)
    }
    if(this.props.recipes.length===0){
        this.setState({
            showMessage:true
        })
    }
  }
  render(){
    if (this.state.redirect) {
      return <Redirect to="/error" />
    }
    return(
      <div className="RecipesTable">
        <h3>&nbsp;<a onClick={this.handleShow} className="btn btn-success pull-right"> Add Recipe</a>
        <RecipeModal handleClose={this.handleClose} show={this.state.show} selected={this.state.selected} recipeData={this.state.recipeData} message={this.props.message} categories={this.props.categories} modalTitle={this.state.modalTitle} handleChange={this.handeChange} handleRecipeUpdate={this.handleRecipeUpdate} addRecipe={this.addRecipe}/>
            <div className="col-xs-7 col-sm-4 pull-right">
                <div className="input-group mb-2 mb-sm-0">
                    <div className="input-group-addon">Search</div>
                    <input type="text" className="form-control" onChange={this.handleRecipeSearch} onKeyUp={this.handleRecipeSearch} placeholder="Enter your search key words here!" id="search"/>
                </div>
            </div>
        </h3><br/>
        {this.props.loading
          ?<center id="loader"><i className="fa fa-spinner fa-pulse fa-4x fa-fw"></i></center>
          :<div>
            {this.props.recipes.length > 0
              ? <div>
                  {this.props.recipes.map((recipe, index) =>
                      <RecipeTableRows id={index + 1} key={recipe.recipe_id}{...recipe} deleteHandler={this.deleteHandler} handleEditData={this.handleEditData} />
                  )}
                  <Pagination recipeData={this.props} nextPage={this.nextPage} previousPage={this.previousPage}/>
                </div>
              
              : <div className="alert alert-info col-sm-8" id="no-recipes">
                  <p>No recipes at the moment!<a onClick={this.handleShow} className="btn btn-primary pull-right"> Add Recipe</a></p>
                </div>
            }
          </div>
        }
      </div>
    )
}
}

function mapStateToProps(state, ownProps){
    let disableNext = ""
    let disablePrevious = ""
    if(state.recipes.has_next){
      disableNext = "page-item"
    }else{
      disableNext = "page-item disabled" 
    }
    if(state.recipes.has_prev){
      disablePrevious = "page-item"
    }else{
      disablePrevious = "page-item disabled"
    }

  return {
      recipes:state.recipes.recipes,
      page:state.recipes.page,
      total_pages:state.recipes.total_pages,
      has_next:state.recipes.has_next,
      has_prev:state.recipes.has_prev,
      previous_page:state.recipes.previous_page,
      next_page:state.recipes.next_page,
      message:state.recipes.message,
      disablePrevious,
      disableNext,
      loading: state.ajaxCallsInProgress > 0,
      categories: state.categories.categories,
      recipe:state.recipes.recipe
  }
}

function mapDispatchToProps(dispatch){
  return{
    loadRecipes:(url) => dispatch(loadRecipes(url)),
    loadRecipe:(id) => dispatch(loadRecipe(id)),
    loadCategories:(url) => dispatch(loadCategories(url)),
    addRecipe:(data) => dispatch(addRecipe(data)),
    updateRecipe:(id,data) => dispatch(updateRecipe(id,data)),
    deleteRecipe:(id) => dispatch(deleteRecipe(id)),
    
  }
}
export default connect(mapStateToProps, mapDispatchToProps) (RecipesTable);