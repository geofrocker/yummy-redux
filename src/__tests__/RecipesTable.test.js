import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { RecipesTable } from '../components/dashboard/recipesTab/RecipesTable';
import toJson from 'enzyme-to-json';

describe('the Recipe Table component', () => {
  const props = {
    str: {},
    history: {
      push: () => {},
    },
    categories: [{ cat_id: 1, cat_name: 'lorem' }],
    recipes: [{ recipe_id: 1, title: 'Lorem', ingredients: 'Ipsum' }],
    loadRecipes: () => Promise.resolve(),
    loadCategories: () => Promise.resolve(),
    deleteRecipe: () => Promise.resolve(),

  };
  const wrapper = mount(<RecipesTable {...props} />);
  const preventDefault = jest.fn();

  it('renders the RecipesTable class', () => {
    expect(wrapper.find('.RecipesTable')).toHaveLength(1);
  });

  it('renders the Modal', () => {
    expect(wrapper.find('Modal')).toHaveLength(2);
  });

  it('renders the Recipe table rows', () => {
    expect(wrapper.find('RecipeTableRows')).toHaveLength(1);
  });

  it('renders the Pagination', () => {
    expect(wrapper.find('Pagination')).toHaveLength(1);
  });

  it('renders ten div jsx elements', () => {
    expect(wrapper.find('div')).toHaveLength(10);
  });
  it('renders search bar', () => {
    expect(wrapper.find('#search')).toHaveLength(1);
  });

  it('renders pagination', () => {
    wrapper.setState({ data: [{ a: 'general' }] });
    expect(wrapper.find('.pagination')).toHaveLength(1);
    expect(wrapper.find('.page-link')).toHaveLength(3);
  });

  it('renders search input', () => {
    expect(wrapper.find('#search')).toHaveLength(1);
  });

  it(' handle going back to previous page ', () => {
    wrapper.find('#prevPage').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it(' handle going back to next page ', () => {
    wrapper.find('#nextPage').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it(' handles search feature ', () => {
    wrapper.find('#search').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it(' handle deleting the recipe ', () => {
    wrapper.find('#delete').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
