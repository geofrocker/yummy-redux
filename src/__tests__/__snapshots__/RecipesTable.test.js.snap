// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Recipe Table component  handle deleting the recipe  1`] = `
<RecipesTable
  categories={
    Array [
      Object {
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteRecipe={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="RecipesTable"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addRecipe"
        onClick={[Function]}
      >
         Add Recipe
      </a>
      <RecipeModal
        addRecipe={[Function]}
        categories={
          Array [
            Object {
              "cat_id": 1,
              "cat_name": "lorem",
            },
          ]
        }
        handleChange={[Function]}
        handleClose={[Function]}
        handleRecipeUpdate={[Function]}
        modalTitle="Add Recipe"
        recipeData={
          Object {
            "category": "",
            "category_rel": "",
            "ingredients": "",
            "status": "",
            "steps": "",
            "title": "",
          }
        }
        selected=""
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </RecipeModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="search"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <RecipeTableRows
          deleteHandler={[Function]}
          handleEditData={[Function]}
          id={1}
          ingredients="Ipsum"
          key="1"
          recipe_id={1}
          title="Lorem"
        >
          <div
            className="col-md-4"
          >
            <Thumbnail
              alt="242x200"
              bsClass="thumbnail"
              src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
            >
              <div
                className="thumbnail"
              >
                <img
                  alt="242x200"
                  src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
                />
                <div
                  className="caption"
                >
                  <h3>
                    Lorem
                  </h3>
                  <p>
                    Ipsum...
                  </p>
                  <p>
                    <a
                      className="btn btn-default"
                      href="/recipe/1"
                      id="view"
                    >
                      <span
                        className="fa fa-eye"
                      >
                         View
                      </span>
                    </a>
                     
                    <button
                      className="btn btn-primary"
                      id="edit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger"
                      id="delete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </p>
                </div>
              </div>
            </Thumbnail>
          </div>
        </RecipeTableRows>
        <Pagination
          nextPage={[Function]}
          previousPage={[Function]}
          recipeData={
            Object {
              "categories": Array [
                Object {
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteRecipe": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
        >
          <center>
            <div
              className="col-xs-12 col-sm-12 pull-right"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="prevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="nextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</RecipesTable>
`;

exports[`the Recipe Table component  handle going back to next page  1`] = `
<RecipesTable
  categories={
    Array [
      Object {
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteRecipe={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="RecipesTable"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addRecipe"
        onClick={[Function]}
      >
         Add Recipe
      </a>
      <RecipeModal
        addRecipe={[Function]}
        categories={
          Array [
            Object {
              "cat_id": 1,
              "cat_name": "lorem",
            },
          ]
        }
        handleChange={[Function]}
        handleClose={[Function]}
        handleRecipeUpdate={[Function]}
        modalTitle="Add Recipe"
        recipeData={
          Object {
            "category": "",
            "category_rel": "",
            "ingredients": "",
            "status": "",
            "steps": "",
            "title": "",
          }
        }
        selected=""
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </RecipeModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="search"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <RecipeTableRows
          deleteHandler={[Function]}
          handleEditData={[Function]}
          id={1}
          ingredients="Ipsum"
          key="1"
          recipe_id={1}
          title="Lorem"
        >
          <div
            className="col-md-4"
          >
            <Thumbnail
              alt="242x200"
              bsClass="thumbnail"
              src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
            >
              <div
                className="thumbnail"
              >
                <img
                  alt="242x200"
                  src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
                />
                <div
                  className="caption"
                >
                  <h3>
                    Lorem
                  </h3>
                  <p>
                    Ipsum...
                  </p>
                  <p>
                    <a
                      className="btn btn-default"
                      href="/recipe/1"
                      id="view"
                    >
                      <span
                        className="fa fa-eye"
                      >
                         View
                      </span>
                    </a>
                     
                    <button
                      className="btn btn-primary"
                      id="edit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger"
                      id="delete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </p>
                </div>
              </div>
            </Thumbnail>
          </div>
        </RecipeTableRows>
        <Pagination
          nextPage={[Function]}
          previousPage={[Function]}
          recipeData={
            Object {
              "categories": Array [
                Object {
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteRecipe": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
        >
          <center>
            <div
              className="col-xs-12 col-sm-12 pull-right"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="prevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="nextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</RecipesTable>
`;

exports[`the Recipe Table component  handle going back to previous page  1`] = `
<RecipesTable
  categories={
    Array [
      Object {
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteRecipe={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="RecipesTable"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addRecipe"
        onClick={[Function]}
      >
         Add Recipe
      </a>
      <RecipeModal
        addRecipe={[Function]}
        categories={
          Array [
            Object {
              "cat_id": 1,
              "cat_name": "lorem",
            },
          ]
        }
        handleChange={[Function]}
        handleClose={[Function]}
        handleRecipeUpdate={[Function]}
        modalTitle="Add Recipe"
        recipeData={
          Object {
            "category": "",
            "category_rel": "",
            "ingredients": "",
            "status": "",
            "steps": "",
            "title": "",
          }
        }
        selected=""
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </RecipeModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="search"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <RecipeTableRows
          deleteHandler={[Function]}
          handleEditData={[Function]}
          id={1}
          ingredients="Ipsum"
          key="1"
          recipe_id={1}
          title="Lorem"
        >
          <div
            className="col-md-4"
          >
            <Thumbnail
              alt="242x200"
              bsClass="thumbnail"
              src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
            >
              <div
                className="thumbnail"
              >
                <img
                  alt="242x200"
                  src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
                />
                <div
                  className="caption"
                >
                  <h3>
                    Lorem
                  </h3>
                  <p>
                    Ipsum...
                  </p>
                  <p>
                    <a
                      className="btn btn-default"
                      href="/recipe/1"
                      id="view"
                    >
                      <span
                        className="fa fa-eye"
                      >
                         View
                      </span>
                    </a>
                     
                    <button
                      className="btn btn-primary"
                      id="edit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger"
                      id="delete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </p>
                </div>
              </div>
            </Thumbnail>
          </div>
        </RecipeTableRows>
        <Pagination
          nextPage={[Function]}
          previousPage={[Function]}
          recipeData={
            Object {
              "categories": Array [
                Object {
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteRecipe": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
        >
          <center>
            <div
              className="col-xs-12 col-sm-12 pull-right"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="prevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="nextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</RecipesTable>
`;

exports[`the Recipe Table component  handles search feature  1`] = `
<RecipesTable
  categories={
    Array [
      Object {
        "cat_id": 1,
        "cat_name": "lorem",
      },
    ]
  }
  deleteRecipe={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loadCategories={[Function]}
  loadRecipes={[Function]}
  recipes={
    Array [
      Object {
        "ingredients": "Ipsum",
        "recipe_id": 1,
        "title": "Lorem",
      },
    ]
  }
  str={Object {}}
>
  <div
    className="RecipesTable"
  >
    <h3>
       
      <a
        className="btn btn-success pull-right"
        id="addRecipe"
        onClick={[Function]}
      >
         Add Recipe
      </a>
      <RecipeModal
        addRecipe={[Function]}
        categories={
          Array [
            Object {
              "cat_id": 1,
              "cat_name": "lorem",
            },
          ]
        }
        handleChange={[Function]}
        handleClose={[Function]}
        handleRecipeUpdate={[Function]}
        modalTitle="Add Recipe"
        recipeData={
          Object {
            "category": "",
            "category_rel": "",
            "ingredients": "",
            "status": "",
            "steps": "",
            "title": "",
          }
        }
        selected=""
        show={false}
      >
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropClassName="modal-backdrop"
            backdropTransition={[Function]}
            containerClassName="modal-open"
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onEntering={[Function]}
            onExited={[Function]}
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
            transition={[Function]}
          />
        </Modal>
      </RecipeModal>
      <div
        className="col-xs-7 col-sm-4 pull-right"
      >
        <div
          className="input-group mb-2 mb-sm-0"
        >
          <div
            className="input-group-addon"
          >
            Search
          </div>
          <input
            className="form-control"
            id="search"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Enter your search key words here!"
            type="text"
          />
        </div>
      </div>
    </h3>
    <br />
    <div>
      <div>
        <RecipeTableRows
          deleteHandler={[Function]}
          handleEditData={[Function]}
          id={1}
          ingredients="Ipsum"
          key="1"
          recipe_id={1}
          title="Lorem"
        >
          <div
            className="col-md-4"
          >
            <Thumbnail
              alt="242x200"
              bsClass="thumbnail"
              src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
            >
              <div
                className="thumbnail"
              >
                <img
                  alt="242x200"
                  src="https://mperumpy.sirv.com/Images/brooke-lark-158017.jpg"
                />
                <div
                  className="caption"
                >
                  <h3>
                    Lorem
                  </h3>
                  <p>
                    Ipsum...
                  </p>
                  <p>
                    <a
                      className="btn btn-default"
                      href="/recipe/1"
                      id="view"
                    >
                      <span
                        className="fa fa-eye"
                      >
                         View
                      </span>
                    </a>
                     
                    <button
                      className="btn btn-primary"
                      id="edit"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-edit"
                      >
                         Edit
                      </span>
                    </button>
                     
                    <button
                      className="btn btn-danger"
                      id="delete"
                      onClick={[Function]}
                    >
                      <span
                        className="fa fa-trash"
                      >
                         Delete
                      </span>
                    </button>
                  </p>
                </div>
              </div>
            </Thumbnail>
          </div>
        </RecipeTableRows>
        <Pagination
          nextPage={[Function]}
          previousPage={[Function]}
          recipeData={
            Object {
              "categories": Array [
                Object {
                  "cat_id": 1,
                  "cat_name": "lorem",
                },
              ],
              "deleteRecipe": [Function],
              "history": Object {
                "push": [Function],
              },
              "loadCategories": [Function],
              "loadRecipes": [Function],
              "recipes": Array [
                Object {
                  "ingredients": "Ipsum",
                  "recipe_id": 1,
                  "title": "Lorem",
                },
              ],
              "str": Object {},
            }
          }
        >
          <center>
            <div
              className="col-xs-12 col-sm-12 pull-right"
              id="pagination"
            >
              <ul
                className="pagination"
              >
                <li>
                  <a
                    className="page-link"
                    id="prevPage"
                    onClick={[Function]}
                  >
                    Previous
                  </a>
                </li>
                <li>
                  <a
                    className="page-link"
                    id="nextPage"
                    onClick={[Function]}
                  >
                    Next
                  </a>
                  <a
                    className="page-link"
                  >
                    Showing 
                     of 
                  </a>
                </li>
              </ul>
            </div>
          </center>
        </Pagination>
      </div>
    </div>
  </div>
</RecipesTable>
`;
